yt/frontends/gadget/fields.py

```diff
@@ -122,6 +128,22 @@ def _h_density(field, data):
 
         return ["H"] + metal_names[:-1]
 
+    def _check_whitelist_species_fields(self, ptype):
+        species_names = []
+        species_nums = []
+        for field in self.ds.field_list:
+            if field[0] == ptype:
+                if field[1].endswith("_fraction") or field[1].endswith("_density"):
+                    if field[1] in sph_whitelist_fields:
+                        symbol = field[1].split("_")[0]
```

Comment:
Ok I see a potential bug here.
For now, because items in `sph_whitelist_fields` that end in `"_fraction"` or `"_density"` all contain exactly 1 underscore, there is no difference between the following
```python
symbol = field[1].split("_")[0]          # your implementation
symbol, _, _ = field[1].partition("_")   # refactor 1
symbol, _, _ = field[1].rpartition("_")  # refactor 2
```
However, to avoid creating bugs in the future, we should anticipate now and make sure this still behaves as expected if we suddenly have a (for instance) `"dark_energy_density"` field in that list, and make the implementation as unambiguous as possible.
If you mean to take "everything before the first underscore", then the first proposed refactor would work. Otherwise, if you mean "everything before the *last* underscore" (and I suspect that's the real intention here), then the second refactor would be the more robust solution.

Reply:
Handled